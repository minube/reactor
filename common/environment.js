import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
