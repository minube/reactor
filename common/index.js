import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
