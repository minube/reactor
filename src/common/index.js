import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
