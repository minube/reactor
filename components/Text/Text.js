import { bool, string } from 'prop-types';
import React from 'react';
import { StyleSheet, Text as NativeText } from 'react-native';

import { layout } from '../../common';
import styles from './Text.style';

const Text = ({
  accent, bold, color, large, lighten, primary, small, tiny,
  layout: { TEXT } = layout(), // eslint-disable-line
  ...inherit
}) => (
  <NativeText
    {...inherit}
    style={StyleSheet.flatten([
      styles.container,

      bold && styles.bold,
      tiny && { ...TEXT.TINY },
      small && { ...TEXT.SMALL },
      large && { ...TEXT.LARGE },

      inherit.style,
      lighten && styles.lighten,
      primary && styles.primary,
      accent && styles.accent,
      color && { color },
    ])}
  />
);

Text.propTypes = {
  accent: bool,
  bold: bool,
  color: string,
  large: bool,
  lighten: bool,
  primary: bool,
  small: bool,
  tiny: bool,
};

Text.defaultProps = {
  accent: false,
  bold: false,
  color: undefined,
  large: false,
  lighten: false,
  primary: false,
  small: false,
  tiny: false,
};

export default Text;
