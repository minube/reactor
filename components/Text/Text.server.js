import { createElement } from 'react';

import { THEME } from '../../common';

const { COLOR } = THEME;
const STYLE = { color: COLOR.TEXT, marginTop: 0, marginBottom: 0 };

export default ({
  color, title, subtitle, large, small, tiny, ...inherit
}) => {
  let tag = 'p';

  if (title) tag = 'h1';
  else if (subtitle) tag = 'h2';
  else if (large) tag = 'h3';
  else if (small || tiny) tag = 'small';

  return createElement(tag, { style: { ...STYLE, color } }, inherit.children);
};
