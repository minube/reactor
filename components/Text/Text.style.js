import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { COLOR, FONT, UNIT } = THEME;

const HEADLINE = {
  fontFamily: FONT.FAMILY_SECONDARY,
  fontWeight: FONT.WEIGHT.BOLD,
};

const SUBTITLE = {
  fontWeight: FONT.WEIGHT.SEMIBOLD,
};

export default StyleSheet.create({
  container: {
    color: COLOR.TEXT,
    fontFamily: FONT.FAMILY,
    fontWeight: FONT.WEIGHT.REGULAR,
    textAlignVertical: 'center',
  },

  headline0: {
    ...HEADLINE, fontSize: UNIT * 12, lineHeight: UNIT * 12, letterSpacing: UNIT * -0.5,
  },
  headline1: {
    ...HEADLINE, fontSize: UNIT * 9.4, lineHeight: UNIT * 9.4, letterSpacing: UNIT * -0.392,
  },
  headline2: {
    ...HEADLINE, fontSize: UNIT * 6.0, lineHeight: UNIT * 6.0, letterSpacing: UNIT * -0.25,
  },
  headline3: {
    ...HEADLINE, fontSize: UNIT * 4.8, lineHeight: UNIT * 4.8, letterSpacing: UNIT * -0.2,
  },
  headline4: {
    ...HEADLINE, fontSize: UNIT * 3.2, lineHeight: UNIT * 3.4, letterSpacing: UNIT * -0.133,
  },
  headline5: {
    ...HEADLINE, fontSize: UNIT * 2.4, lineHeight: UNIT * 2.6, letterSpacing: UNIT * -0.1,
  },
  headline6: {
    ...HEADLINE, fontSize: UNIT * 2.0, lineHeight: UNIT * 2.2, letterSpacing: parseFloat((UNIT * -0.083).toFixed(2), 10),
  },

  subtitle1: { ...SUBTITLE, fontSize: UNIT * 2.0, lineHeight: UNIT * 2.2 },
  subtitle2: { ...SUBTITLE, fontSize: UNIT * 1.6, lineHeight: UNIT * 1.8 },
  subtitle3: { ...SUBTITLE, fontSize: UNIT * 1.4, lineHeight: UNIT * 1.6 },

  body1: { fontSize: UNIT * 1.6, lineHeight: UNIT * 3.0 },
  body2: { fontSize: UNIT * 1.4, lineHeight: UNIT * 2.4 },
  body3: { fontSize: UNIT * 1.2, lineHeight: UNIT * 1.7 },

  caption1: { fontSize: UNIT * 1.4, lineHeight: UNIT * 1.8 },
  caption2: { fontSize: UNIT * 1.2, lineHeight: UNIT * 1.6 },

  lighter: { fontWeight: FONT.WEIGHT.LIGHTER },

  semibold: { fontWeight: FONT.WEIGHT.SEMIBOLD },

  bold: { fontWeight: FONT.WEIGHT.BOLD },

  lighten: { color: COLOR.TEXT_LIGHTEN },
});
