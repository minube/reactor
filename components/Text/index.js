import { ENV } from '../../common';
import Default from './Text';

const Loading = ENV.IS_SERVER ? require('./Text.server').default : Default;

export default Loading;
