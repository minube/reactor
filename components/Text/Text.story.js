import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  text, color, boolean, number,
} from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Text from './Text';

const LOREM = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.';
const LIPSUM = "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
const style = {
  backgroundColor: 'rgba(0,255,0,0.25)',
  padding: 10,
};

storiesOf('✅ Text', module)
  .add('default', () => (
    <Text>{LIPSUM}</Text>
  ))
  .add('headline (0)', () => (
    <Text headline level={0}>{LOREM}</Text>
  ))
  .add('headline (1)', () => (
    <Text headline level={1}>{LOREM}</Text>
  ))
  .add('headline (2)', () => (
    <Text headline level={2}>{LOREM}</Text>
  ))
  .add('headline (3)', () => (
    <Text headline level={3}>{LOREM}</Text>
  ))
  .add('headline (4)', () => (
    <Text headline level={4}>{LOREM}</Text>
  ))
  .add('headline (5)', () => (
    <Text headline level={5}>{LOREM}</Text>
  ))
  .add('headline (6)', () => (
    <Text headline level={6}>{LOREM}</Text>
  ))
  .add('subtitle (1)', () => (
    <Text subtitle level={1}>{LIPSUM}</Text>
  ))
  .add('subtitle (2)', () => (
    <Text subtitle level={2}>{LIPSUM}</Text>
  ))
  .add('subtitle (3)', () => (
    <Text subtitle level={3}>{LIPSUM}</Text>
  ))
  .add('body (1)', () => (
    <Text body level={1}>{LIPSUM}</Text>
  ))
  .add('body (2)', () => (
    <Text body level={2}>{LIPSUM}</Text>
  ))
  .add('body (3)', () => (
    <Text body level={3}>{LIPSUM}</Text>
  ))
  .add('caption (1)', () => (
    <Text body caption level={1}>{LIPSUM}</Text>
  ))
  .add('caption (2)', () => (
    <Text body caption level={2}>{LIPSUM}</Text>
  ))
  .add('lighten', () => (
    <Text lighten>{LIPSUM}</Text>
  ))
  .add('color', () => (
    <Text color={THEME.COLOR.ERROR}>{LIPSUM}</Text>
  ))
  .add('numberOfLines', () => (
    <Text numberOfLines={1}>{LIPSUM}</Text>
  ))
  .add('style', () => (
    <Text style={style}>Customized Text</Text>
  ))
  .add('🏀 Playground', () => (
    <Text
      headline={boolean('headline', false)}
      subtitle={boolean('subtitle', false)}
      body={boolean('body', true)}
      caption={boolean('caption', false)}
      level={number('level', 1)}
      lighten={boolean('lighten', false)}
      color={color('color', THEME.COLOR.TEXT)}
      numberOfLines={number('numberOfLines', 1)}
    >
      {text('content', LOREM)}
    </Text>
  ));
