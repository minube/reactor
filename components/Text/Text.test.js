import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Text from './Text';

describe('<Text>', () => {
  it('renders', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {tiny}', () => {
    const tree = renderer.create(<Text tiny>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {small}', () => {
    const tree = renderer.create(<Text small>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {large}', () => {
    const tree = renderer.create(<Text large>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<Text title>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {lighter}', () => {
    const tree = renderer.create(<Text lighter>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {semibold}', () => {
    const tree = renderer.create(<Text semibold>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {bold}', () => {
    const tree = renderer.create(<Text bold>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {italic}', () => {
    const tree = renderer.create(<Text italic>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {primary}', () => {
    const tree = renderer.create(<Text primary>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {accent}', () => {
    const tree = renderer.create(<Text accent>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {lighten}', () => {
    const tree = renderer.create(<Text lighten>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('when {color}', () => {
    const tree = renderer.create(<Text color={THEME.COLOR.ERROR}>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {numberOfLines}', () => {
    const tree = renderer.create(<Text numberOfLines={2}>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Text style={{ fontSize: THEME.FONT.SIZE.LARGE, color: 'orange' }}>Hello World</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Text />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

