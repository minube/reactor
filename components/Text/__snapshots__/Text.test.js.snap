// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
/>
`;

exports[`<Text> when {accent} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#f69a37",
      "fontFamily": undefined,
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {bold} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 16,
      "fontWeight": "bold",
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {color} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#ef5160",
      "fontFamily": undefined,
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {italic} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 16,
      "fontStyle": "italic",
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {large} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 18,
      "lineHeight": 23.400000000000002,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {lighten} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#9b9b9b",
      "fontFamily": undefined,
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {lighter} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 16,
      "fontWeight": "300",
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {primary} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#10c0bc",
      "fontFamily": undefined,
      "fontSize": 16,
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {semibold} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 16,
      "fontWeight": "600",
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {small} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 14,
      "lineHeight": 16.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {style} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "orange",
      "fontFamily": undefined,
      "fontSize": 20,
      "lineHeight": 20.8,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {tiny} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 12,
      "lineHeight": 13.799999999999999,
    }
  }
>
  Hello World
</Text>
`;

exports[`<Text> when {title} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4a4a4a",
      "fontFamily": undefined,
      "fontSize": 30,
      "lineHeight": 39,
    }
  }
>
  Hello World
</Text>
`;
